components/grade-table.js

```diff
@@ -11,21 +12,65 @@ class GradeTable{
         for(var i=0; i<grades.length; i++){
             var gradesIndex = grades[i];
 
-            var tableRow = document.createElement('tr');
+            this.renderGradeRow(gradesIndex,this.deleteGrade);
 
-            var studentName = document.createElement('td');
-            studentName.textContent = gradesIndex.name;
-            tableRow.appendChild(studentName);
+            if(grades.length === 0){
+                var pEl = document.getElementById('no-grades');
+                pEl.classList.remove("d-none");
+            } 
```

Comment:
There are a couple issues here:
- Rather than querying the whole `document` again, why not utilize the reference to the `#no-grades` element saved in the `constructor`?
- This probably only needs to run once each time `.updateGrades(...)` is called, not once for every row.
- Related to the previous issue, if `grades` is empty, this doesn't run at all.
- What happens if the `#no-grades` message is showing, then a new grade is added? I suspect it'll need to be hidden again.